common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockHandler.java

```diff
@@ -209,4 +190,52 @@ private ShuffleMetrics() {
     }
   }
 
+  private class ManagedBufferIterator implements Iterator<ManagedBuffer> {
+
+    private int index = 0;
+    private String appId;
+    private String execId;
+    private String shuffleId;
+    // An array containing mapId and reduceId pairs.
+    private int[][] mapIdAndReduceIds;
+
+    ManagedBufferIterator(String appId, String execId, String[] blockIds) {
+      this.appId = appId;
+      this.execId = execId;
+      String[] blockId0Parts = blockIds[0].split("_");
+      if (blockId0Parts.length < 4) {
+        throw new IllegalArgumentException("Unexpected block id format: " + blockIds[0]);
+      } else if (!blockId0Parts[0].equals("shuffle")) {
```

Comment:
We have some kinds of `BlockId`, I guess it's better to have a check here and we can parse the blockId correctly.